### ISSUE TITLE
Misplaced semicolon in arbitrarily ordered keywords.

### ISSUE THREAD

**Issue by haberdashPI** [domluna/JuliaFormatter.jl#625]
```julia
foo(x, bar=0, y)
foo(x, y, bar=0)
```

Becomes:

```julia
foo(x; bar=0, y)
foo(x, y; bar=0)
```

😭

**Comment by domluna** [comment]
how is the semicolon misplaced?

**Comment by haberdashPI** [comment]
Observe:

```julia
julia> foo(a, b=5; bar, y=2) = y*(a+b)
foo (generic function with 2 methods)

julia> foo(x, bar=0, y)
6

julia> foo(x, y, bar=0)
6

julia> foo(x; bar=0, y)
12

julia> foo(x, y; bar=0)
6
```

To elaborate: in julia `foo(x, bar=0, y)` is semantically equivalent to `foo(x, y, bar=0)`. That is, without the semilcolon, keyword argument location doesn't matter, even if it occurs before a position argument.

In the first line of the reformatted example, `y` has become a keyword argument instead of a positional argument, changing how `y` is passed to `foo`.

The correct reformatting would be:

```julia
foo(x, y; bar=0)
foo(x, y; bar=0)
```

**Comment by domluna** [comment]
ok ic (i think). This is a super confusing way to write a function though.

**Comment by haberdashPI** [comment]
Agreed: I don't think place keyword arguments between the ordered arguments would be considered good practice. However, it does seem like a problem that reformatting can introduce bugs / errors into a program.

**Comment by penelopeysm** [comment]
sigh - Julia syntax is weird with kwargs, i would have been happy if kwargs always had to be separated with semicolon...

**Comment by penelopeysm** [comment]
Will be fixed in next release.

**Comment by penelopeysm** [comment]
#1141
